src/components/donation-flow/DonationFlowStatusPage.tsx

```diff
@@ -0,0 +1,201 @@
+import { useEffect, useRef, useState } from 'react'
+import Link from 'next/link'
+import { useRouter } from 'next/router'
+import { useSession } from 'next-auth/react'
+import { Form, Formik, FormikProps } from 'formik'
+import {
+  Box,
+  CircularProgress,
+  Stack,
+  Typography,
+  Card,
+  CardContent,
+  CardActionArea,
+  Unstable_Grid2 as Grid2,
+} from '@mui/material'
+import { Email } from '@mui/icons-material'
+
+import { stripe } from 'service/stripeClient'
+import { routes } from 'common/routes'
+
+import { useViewCampaign } from 'common/hooks/campaigns'
+import CenteredSpinner from 'components/common/CenteredSpinner'
+import FormTextField from 'components/common/form/FormTextField'
+import SocialShareListButton from 'components/common/SocialShareListButton'
+import SubmitButton from 'components/common/form/SubmitButton'
+import theme from 'common/theme'
+
+import SuccessGraphic from './icons/SuccessGraphic'
+import { DonationFormPaymentStatus } from './helpers/types'
+import DonationFlowLayout from './DonationFlowLayout'
+import StepSplitter from './common/StepSplitter'
+import { useMutation } from '@tanstack/react-query'
+import { createDonationWish } from 'service/donationWish'
+import { AlertStore } from 'stores/AlertStore'
+import { useCurrentPerson } from 'common/util/useCurrentPerson'
+
+function LinkCard({ href, text }: { href: string; text: string }) {
+  return (
+    <Card sx={{ backgroundColor: '#eee' }}>
+      <CardActionArea sx={{ p: 3 }} LinkComponent={Link} href={href}>
+        <CardContent>
+          <Typography textAlign="center" variant="h6" component="div">
+            {text}
+          </Typography>
+        </CardContent>
+      </CardActionArea>
+    </Card>
+  )
+}
+
+export default function DonationFlowStatusPage({ slug }: { slug: string }) {
+  const { data, isLoading } = useViewCampaign(slug)
+  if (isLoading || !data) return <CenteredSpinner size="2rem" />
```

Comment:
We break the rules of hooks with an early return before some of the hooks. It's better to create a component wrapper that will render this component only in certain conditions.

Reply:
That is true. I will change it just in case.
Though considering this request is prefetched this never turns out to be a problem, because this condition is never true, even on first render. Otherwise I probably would have gotten the `rules-of-hooks` error